palette/src/gradient.rs

```diff
@@ -92,7 +92,14 @@ impl<C: Mix + Clone> Gradient<C> {
         min_color.mix(max_color, factor)
     }
 
-    ///Take `n` evenly spaced colors from the gradient, as an iterator.
+    ///Take `n` evenly spaced colors from the gradient, as an iterator. Now
+    ///inclsuive of both ends of the gradient. Points are spaced 1/(n-1)
+    ///distance apart for n > 1. There are n-2 intermediate points
+    ///between start and end for n > 2.
+    ///
+    ///For example, `take(5)` will include point 0.0 of the gradient, three
+    ///intermediate colors, and point 1.0 spaced apart at 1/4 the distance
+    ///between colors 0.0 and 1.0 on the gradient.
```

Comment:
I would suggest shortening the first paragraph to something like

> Take `n` evenly spaced colors from the gradient, as an iterator. The iterator includes both ends of the gradient, for `n > 1`, or just the lower end of the gradient for `n = 0`.

And then add a small example that illustrates the second paragraph:

```rust
let gradient = Gradient::new(vec![
    LinSrgb::new(1.0, 1.0, 0.0),
    LinSrgb::new(0.0, 0.0, 1.0),
]);

let colors = gradient.take(5).collect();
assert_relative_eq!(colors, vec![
    LinSrgb::new(1.0, 1.0, 0.0),
    LinSrgb::new(0.75, 0.75, 0.25),
    LinSrgb::new(0.5, 0.5, 0.5),
    LinSrgb::new(0.25, 0.25, 0.75),
    LinSrgb::new(0.0, 0.0, 1.0),
]);
```